backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/TransferDiskImageCommand.java

```diff
@@ -458,7 +479,9 @@ protected Map<String, Pair<String, String>> getExclusiveLocks() {
             // StartVmBackup should handle locks
             return locks;
         }
-        if (getDiskImage() != null) {
+
+        // If the disk is a thin template overlay we only lock with a shared lock
+        if (getDiskImage() != null && !isThinTemplate(getDiskImage())) {
```

Comment:
what's the reason for not locking cloned disks also with shared locks?

Reply:
cloned disks have a different id, no? they should be affected by this

Reply:
yes, we generate a different ID from them but regardless of this issue - if all we care of is that the disk is not removed or moved or converted (and such) then we don't have to create an exclusive lock for the disks at all
another way to think about it is to say exclusive locks = write locks and shared locks = read locks (although it's not always true with our engine locks) so if all we do when downloading a disk is read them, then a shared lock seems be more appropriate

Reply:
ah ok, I thought the comment was about an issue... I definitely agree there is no reason to use an exclusive lock here